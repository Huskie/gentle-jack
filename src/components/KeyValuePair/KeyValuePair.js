import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'react-emotion';

const StyledWrapper = Styled('div')`

`;

const StyledKey = Styled('span')`
    font-weight: 700;
    margin-right: 0.25rem;
`;

const KeyValuePair = ({ theKey, value }) => (
  <StyledWrapper>
    <StyledKey>
      {theKey}
:
    </StyledKey>
    <span>{value}</span>
  </StyledWrapper>
);

KeyValuePair.propTypes = {
  theKey: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
};

KeyValuePair.defaultProps = {};

export { KeyValuePair };
