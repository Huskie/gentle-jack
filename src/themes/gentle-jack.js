const theme = {
  borderRadius: '4px',
  colours: {
    alpha: '#2742b8',
    beta: '#222',
    gamma: '#06f',
    delta: '#fff',
    epsilon: '#1b6cff',
    zeta: '#ffd561',
    eta: '#4048ef',
  },
  gradients: {
    alpha: '#4048ef, #2742b8',
  },
  spacing: {
    componentMargin: '16px',
  },
  sizing: {
    maxWidth: '1080px',
  },
};

export { theme };
