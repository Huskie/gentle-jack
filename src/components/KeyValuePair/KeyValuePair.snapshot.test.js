import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'emotion-theming';

import { KeyValuePair } from './KeyValuePair';
import { theme } from '../../themes/gentle-jack';

describe('KeyValuePair snapshots:', () => {
  it('component renders with defaults', () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <KeyValuePair theKey="A test key" value="A test value" />
      </ThemeProvider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
