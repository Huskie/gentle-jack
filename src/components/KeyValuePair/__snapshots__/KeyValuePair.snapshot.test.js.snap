// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyValuePair snapshots: component renders with defaults 1`] = `
.emotion-0 {
  font-weight: 700;
  margin-right: 0.25rem;
}

<div
  className="emotion-1"
>
  <span
    className="emotion-0"
  >
    A test key
    :
  </span>
  <span>
    A test value
  </span>
</div>
`;
